### ISSUE TITLE
Confusing BoundsError for JuMPArray

### ISSUE THREAD

**Issue by joehuchette** [JuliaOpt/JuMP.jl#549]
```jl
julia> using JuMP

julia> m = Model();

julia> @defVar(m, x[2:3]);

julia> x[4]
ERROR: BoundsError: attempt to access 2-element Array{JuMP.Variable,1}:
 x[2]
 x[3]
  at index [3]
 in getindex at no file
```

**Comment by mlubin** [comment]
We'll fix this in the JuMPArray rewrite, post 0.10.

**Comment by mlubin** [comment]
Looks like this is solved now?
```
julia> x[4]
ERROR: Failed attempt to index JuMPArray along dimension 1: 4 ∉ 2:3
 in error at error.jl:21
 in getindex at /home/mlubin/.julia/v0.4/JuMP/src/JuMPArray.jl:70
```
